src/averell/readers/ecpa.py

```diff
@@ -0,0 +1,142 @@
+                line_group_list2.append(lg)
+            if lg.get("met"):
+                line_group_list2.append(lg)
+    stanza_list = []
+    line_number = 0
+    for stanza_number, line_group in enumerate(line_group_list2):
+        stanza_type = ""
+        stanza_text = []
+        line_list = []
+        for n, line in enumerate(line_group.findall(f"{NS}l")):
+            line_dict = {}
+            line_id = line.attrib.get(f"{XML_NS}id")
+            line_info = lines_info.get(line_id)
+            line_length = ""
+            met = ""
+            foot = ""
+            metre = ""
+            if line_info:
+                if n == 0:
+                    stanza_type = line_info.get("stanzas", "").get("id", "")
+                syllab = line_info.get("syllab")
+                line_length = int(syllab) if syllab else ""
+                met = line_info.get("met", "").strip("/")
+                foot = line_info.get("foot").get("id", "")
+                metre = line_info.get("footnum").get("id", "")
+                real = line_info.get("real")
+                if real:
+                    manually_checked = True
+                    met = real.strip("/")
+                    foot = line_info.get("realfoot").get("id")
+                    metre = line_info.get("realfootnum").get("id")
+            line_dict.update({
+                "metrical_pattern": met,
+                "line_length": line_length,
+                "foot": foot,
+                "metre": metre,
+            })
+            word_list = []
+            token_list = []
+            for token in line:
+                tag = token.tag
+                if tag == f"{NS}w":
+                    word_list.append({"word_text": token.text})
+                if tag in [f"{NS}w", f"{NS}c", f"{NS}pc"]:
+                    token_list.append(token.text or "")
+            line_text = "".join(token_list).strip()
+            line_dict.update({
+                "line_number": line_number + 1,
+                "line_text": "".join(line_text).strip(),
+                "words": word_list,
+            })
+            line_list.append(line_dict)
+            stanza_text.append(line_text)
+            line_number += 1
+        st = "\n".join(stanza_text)
+        stanza_list.append({
+            "stanza_number": stanza_number + 1,
+            "stanza_type": stanza_type,
+            "lines": line_list,
+            "stanza_text": st,
+        })
+    poem.update({
+        "manually_checked": manually_checked,
+        "stanzas": stanza_list
+    })
+    return poem
+
+
+def get_features(path):
+    """Function to find each poem file and parse it
+
+    :param path: Corpus Path
+    :return: List of poem dicts
+    :rtype: list
+    """
+    authors_file = path / "web" / "resources" / "models" / "authwork_mdp.json"
+    authors = json.loads(authors_file.read_text())
+    xml_files = path / "web" / "works"
+    feature_list = []
+    for filename in xml_files.rglob("*/*.xml"):
```

Comment:
I think `.rglob()` handles recursive folder structures, so you just need `.rblog("*.xml")`

Reply:
Yes, but I don't want to search for xml files within `web/works` folder, but within the subfolders.